.github/scripts/run-tests

```diff
@@ -55,8 +55,19 @@ for TEST in "${TESTS_TO_RUN[@]}"
 do
     LOCAL_SUCCESS=0
     RESULT_XML=$(echo ${TEST} | sed -e "s/${TEST_DIR}/${RESULTS_DIR}/" | sed -e "s/.py/.xml/")
-    pytest --junitxml=${RESULT_XML} ${TEST} || LOCAL_SUCCESS=$?
+
+    # this is a bit messy and brittle, but certain tests
+    # need to be run with specific options
+    if [[ "${TEST}" == *"kernels"* ]] then
+        CUDA_VISIBLE_DEVICES=0 pytest --junitxml=${RESULT_XML} ${TEST} || LOCAL_SUCCESS=$?
+    elif [[ "${TEST}" == *"distributed"* ]] then
+        pytest --forked --junitxml=${RESULT_XML} ${TEST} || LOCAL_SUCCESS=$?
+    else
+        pytest --junitxml=${RESULT_XML} ${TEST} || LOCAL_SUCCESS=$?
+    fi
+
     SUCCESS=$((SUCCESS + LOCAL_SUCCESS))
+
```

Comment:
nit: it might be simpler to factor out the env + flags into variables and only have one call to pytest, e.g.
```
env ${ENV_VARS} pytest ${FLAGS} --junitxml=.... ${TEST} || LOCAL_SUCCESS=$?
```
with ENV_VARS and FLAGS set appropriately

Reply:
i'm okay leaving it as is. it seems the tests need some organization.